apps/dcos_dns/src/dcos_dns_router.erl

```diff
@@ -12,15 +12,23 @@
 %% @doc Resolvers based on a set of "questions"
 -spec(upstreams_from_questions(dns:questions()) ->
     {[upstream()] | internal, binary()}).
-upstreams_from_questions([#dns_query{name=Name}]) ->
-    Labels = dcos_dns_app:parse_upstream_name(Name),
-    find_upstream(Labels);
-upstreams_from_questions([Question|Others]) ->
-    %% There is more than one question. This is beyond our capabilities at the moment
-    dcos_dns_metrics:update([dcos_dns, ignored_questions], length(Others), ?COUNTER),
-    Result = upstreams_from_questions([Question]),
-    lager:debug("~p will be forwarded to ~p", [Others, Result]),
-    Result.
+upstreams_from_questions(Questions) ->
+    UpstreamAnswers = fun (#dns_query{name=Name}) ->
+        Labels = dcos_dns_app:parse_upstream_name(Name),
+        find_upstream(Labels)
+    end,
+    Answers = lists:map(UpstreamAnswers, Questions),
+
+    case lists:usort(Answers) of
+        [Answer] ->
+            Answer;
+        MixedAnswers ->
+            InternalResolver = {internal, <<".">>},
+            lager:notice("DNS question ~p yield multiple results ~p,
```

Comment:
nit: does `lager` do something special to remove extra space chars? The Erlang parser don't:

```
~ $ cat ./multi-line 
#!/usr/bin/env escript
%% -*- erlang -*-

main([]) ->
    io:format("a
        b~n").
~ $ ./multi-line 
a
        b
```

Reply:
Hum, interesting. I saw this message on the screen and it looked fine for me. I probably overlooked it. let me check again

Reply:
Yeah this is broken:
```
01:09:53.720 [notice] A
          B~n
```